tests/lib/rules/require-prop-comment.js

```diff
@@ -0,0 +1,198 @@
+        /** JSDoc comment */
+        a: Number,
+      
+        /* block comment */
+        b: Number,
+      
+        // line comment
+        c: Number,
+      })
+      `,
+      options: [{ type: 'any' }]
+    }
+  ],
+  invalid: [
+    {
+      code: `
+      export default defineComponent({
+        props: {
+          // line comment
+          b: Number,
+      
+          /* block comment */
+          c: Number,
+      
+          d: Number,
+        }
+      })
+      `,
+      errors: [
+        {
+          line: 5,
+          column: 11,
+          message: `The "b" property should have a JSDoc comment.`
+        },
+        {
+          line: 8,
+          column: 11,
+          message: `The "c" property should have a JSDoc comment.`
+        },
+        {
+          line: 10,
+          column: 11,
+          message: `The "d" property should have a JSDoc comment.`
+        }
+      ]
+    },
+    {
+      code: `
+      <script setup>
+      const badProps = defineProps({
+        /** JSDoc comment */
+        b: Number,
+      
+        // line comment
+        c: Number,
+      
+        d: Number,
+      })
+      </script>
+      `,
+      options: [{ type: 'block' }],
+      errors: [
+        {
+          line: 5,
+          column: 9,
+          message: 'The "b" property should have a block comment.'
+        },
+        {
+          line: 8,
+          column: 9,
+          message: 'The "c" property should have a block comment.'
+        },
+        {
+          line: 10,
+          column: 9,
+          message: 'The "d" property should have a block comment.'
+        }
+      ]
+    },
+    {
+      code: `
+      <script setup>
+      const badProps = defineProps({
+        /** JSDoc comment */
+        b: Number,
+      
+        /* block comment */
+        c: Number,
+      
+        d: Number,
+      })
+      </script>
+      `,
+      options: [{ type: 'line' }],
+      errors: [
+        {
+          line: 5,
+          column: 9,
+          message: 'The "b" property should have a line comment.'
+        },
+        {
+          line: 8,
+          column: 9,
+          message: 'The "c" property should have a line comment.'
+        },
+        {
+          line: 10,
+          column: 9,
+          message: 'The "d" property should have a line comment.'
+        }
+      ]
+    },
+    {
+      code: `
+      <script setup>
+      const badProps = defineProps({
+        d: Number,
+      })
+      </script>
+      `,
+      options: [{ type: 'any' }],
+      errors: [
+        {
+          line: 4,
+          column: 9,
+          message: `The "d" property should have a comment.`
+        }
+      ]
+    },
+    {
+      code: `
+      new Vue({
+        props: {
+          a: Number
+        }
+      })
+      `,
+      errors: [
+        {
+          line: 4,
+          column: 11,
+          message: 'The "a" property should have a JSDoc comment.'
+        }
+      ]
+    }
```

Comment:
Could you also add test cases for the definition of type-based?

```vue
<script setup lang="ts">
const props = defineProps<{
  /** JSDoc comment */
  a: number,
  /* block comment */
  b: number,
  // line comment
  c: number,

  d: number,
}>()
</script>
```

Reply:
what do you mean?

Reply:
We need to add tests to see if the following definition method also works.
https://vuejs.org/guide/typescript/composition-api.html#typing-component-props

Reply:
![image](https://user-images.githubusercontent.com/80880408/198958484-18bd0e8e-514e-4d85-a6f9-dd385e75325a.png)
![image](https://user-images.githubusercontent.com/80880408/198958491-254257be-ef30-4924-ab1d-1ffa3251686c.png)
Is that right?

Reply:
No. Define props using only TypeScript types.

```vue
<script setup lang="ts">
const props = defineProps<{
  foo: string
  bar?: number
}>()
</script>
```

Reply:
https://vuejs.org/guide/typescript/composition-api.html#typing-component-props:~:text=However%2C%20it%20is%20usually%20more%20straightforward%20to%20define%20props%20with%20pure%20types%20via%20a%20generic%20type%20argument

<img width="751" alt="image" src="https://user-images.githubusercontent.com/16508807/198959822-dd2acdab-9f05-4fff-a6ad-11da29880763.png">

Reply:
ok